ui/src/app/app.component.ts

```diff
@@ -106,11 +94,11 @@ export class AppComponent  implements OnInit {
     }
 
     startVersionWorker(): void {
-        this.stopVersionWorker();
+        this.stopWorker(this.versionWorker, this.versionWorkerSubscription);
         this.versionWorker = new CDSWorker('./assets/worker/web/version.js');
         this.versionWorker.start({});
         this.versionWorker.response().subscribe( msg => {
-            if (msg !== null) {
+            if (msg !== null && msg !== undefined) {
```

Comment:
Not necessary to change but good to know but if you write `if (msg != null)` it catch undefined and null.